src/main/java/org/mifos/processor/bulk/zeebe/worker/InitSubBatchWorker.java

```diff
@@ -72,15 +72,15 @@ public void setup() {
 
             String fileName = subBatches.remove(0);
             SubBatchEntity subBatchEntity = null;
-
-            for (SubBatchEntity subBatch : subBatchEntityList) {
-                if (subBatch.getRequestFile().contains(fileName)) {
-                    subBatchEntity = subBatch;
-                    logger.info("SubBatchEntity found");
+            if (isSplittingEnabled) {
+                for (SubBatchEntity subBatch : subBatchEntityList) {
+                    if (subBatch.getRequestFile().contains(fileName)) {
+                        subBatchEntity = subBatch;
```

Comment:
break this loop once the required element is found.
write a failure case if not found

Reply:
updated